client/js/socket-events/changelog.js

```diff
@@ -4,19 +4,70 @@ const $ = require("jquery");
 const socket = require("../socket");
 const templates = require("../../views");
 
+module.exports = {
+	requestIfNeeded,
+};
+
+// Requests version information if it hasn't been retrieved before (or if it has
+// been removed from the page, i.e. when clicking on "Check now". Displays a
+// loading state until received.
+function requestIfNeeded() {
+	if ($("#version-checker").is(":empty")) {
+		renderVersionChecker({status: "loading"});
+		socket.emit("changelog");
+	}
+}
+
 socket.on("changelog", function(data) {
-	const container = $("#changelog-version-container");
+	let status;
```

Comment:
Can't this be done in the template? I believe thats how I did it initially.

Reply:
No, because if I do this in template, it recreates the whole div instead of changing the content/class, and that loses the CSS transition.

I would also need to nest 2 divs (one with `id="version-checker"` and one with the correct class). And also what's nice is that I don't have server logic in the template. By server logic I mean `if (data.latest)` and `if (data.current.changelog)` to define the status, template is pretty straightforward that way.

I will however group things together, it's silly right now, I set things for the version checker, then changelog, then version checker.